cpp/src/sort/sort.cu

```diff
@@ -73,22 +76,55 @@ std::unique_ptr<column> sorted_order(table_view input,
         *device_table, *device_table,
         d_column_order.data().get(),
         d_null_precedence.data().get());
-    thrust::sort(rmm::exec_policy(stream)->on(stream),
-                 mutable_indices_view.begin<int32_t>(),
-                 mutable_indices_view.end<int32_t>(), comparator);
-
+    if (stable) {
+      thrust::stable_sort(rmm::exec_policy(stream)->on(stream),
+                          mutable_indices_view.begin<int32_t>(),
```

Comment:
size_type ?

Reply:
I was wondering why the rest of this source file uses `int32_t`. I believe I should change it all to `size_type`?

Reply:
We can change all of them to size_type, but I was just concerned about size_type being moved to int64 and index still in int32  kind of scenario. @jrhemstad What's your opinion ?

Reply:
Use `size_type`. I think I just wasn't being as careful back when I first wrote this.